### ISSUE TITLE
does the classification process not use softmax?

### ISSUE THREAD

**Issue by febrianti2602** [IndoNLP/indonlu#46]
I re-read the Forward function for sequence classification and it turns out there is no external application of softmax. Do you really not use softmax, or do you use softmax internally? If you use softmax internally, where is the softmax implemented?

**Comment by febrianti2602** [comment]
@SamuelCahyawijaya

**Comment by SamuelCahyawijaya** [comment]
@febrianti2602 : Why do we need softmax? If we want to get the probability distribution, we can apply softmax to the logits, but in our case:

1) During inference, since we just want to get the most probable class, so we can simply get top-1 of the logits (as softmax is [a monotonic function](https://stackoverflow.com/questions/56096598/is-softmax-used-when-only-the-most-probable-class-will-be-used))

2) During the training, we use the [CrossEntropyLoss](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) which -- as specified in the documentation itself -- accepts logits as its input and performs a log softmax followed by negative log-likelihood to the input logits.

Hope it helps.
